src/common/devcard.ts

```diff
@@ -25,13 +25,19 @@ export const getMostReadTags = async (
   }));
 };
 
-const getFavoriteSourcesLogos = async (
+interface FavoriteSourcesProps {
+  alias: string;
+  limit: number;
+}
+
+export const generateGetFavoriteSources = (
   con: DataSource,
   userId: string,
-): Promise<string[]> => {
-  const sources = await con
+  { alias = 'source', limit = 5 }: Partial<FavoriteSourcesProps> = {},
```

Comment:
Maybe name it `generateFavoriteSourcesQuery` just so its clear it returns a query builder object? I had to look at the usage to figure it does that?